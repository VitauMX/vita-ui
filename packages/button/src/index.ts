export * from './ButtonPrimary';
export * from './ButtonTertiary';
export * from './ButtonSecondary';
export * from './ButtonText';
