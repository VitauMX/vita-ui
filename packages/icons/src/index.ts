export * from './Icon';
export * from './createIcon';
export * from './icons/';
