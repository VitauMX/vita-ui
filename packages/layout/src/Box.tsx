import styled from 'styled-components';
import {
  space,
  color,
  layout,
  flex,
  SpaceProps,
  ColorProps,
  LayoutProps,
  FlexProps,
  flexGrow,
  FlexGrowProps,
  flexShrink,
  FlexShrinkProps,
  flexBasis,
  FlexBasisProps,
  justifySelf,
  JustifySelfProps,
  alignSelf,
  AlignSelfProps,
  order,
  OrderProps,
  compose,
} from 'styled-system';
import { ComponentProps } from './utils/types';

const flexItem = compose(
  flexGrow,
  flexShrink,
  flexBasis,
  justifySelf,
  alignSelf,
  order
);

interface StyledBoxProps
  extends SpaceProps,
    ColorProps,
    LayoutProps,
    FlexProps,
    FlexGrowProps,
    FlexShrinkProps,
    FlexBasisProps,
    JustifySelfProps,
    AlignSelfProps,
    OrderProps {}

export const Box = styled.div<StyledBoxProps>`
  ${space}
  ${color}
  ${layout}
  ${flex}
  ${flexItem}
`;

export type BoxProps = ComponentProps<typeof Box>;
