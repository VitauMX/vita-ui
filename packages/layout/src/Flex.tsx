import styled from 'styled-components';
import {
  alignItems,
  AlignItemsProps,
  alignContent,
  AlignContentProps,
  justifyItems,
  JustifyItemsProps,
  justifyContent,
  JustifyContentProps,
  flexWrap,
  FlexWrapProps,
  flexDirection,
  FlexDirectionProps,
  compose,
} from 'styled-system';
import { ComponentProps } from './utils/types';

const flex = compose(
  alignItems,
  alignContent,
  justifyItems,
  justifyContent,
  flexWrap,
  flexDirection
);

interface StyledFlexProps
  extends JustifyItemsProps,
    AlignItemsProps,
    JustifyContentProps,
    FlexWrapProps,
    FlexDirectionProps,
    AlignContentProps {}

export const Flex = styled.div<StyledFlexProps>`
  display: flex;
  ${flex}
`;

export type FlexProps = ComponentProps<typeof Flex>;
