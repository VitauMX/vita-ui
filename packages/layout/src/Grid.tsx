import styled from 'styled-components';
import {
  grid as systemGrid,
  GridProps as SystemGridProps,
  justifyItems,
  JustifyItemsProps,
  alignItems,
  AlignItemsProps,
  justifyContent,
  JustifyContentProps,
  alignContent,
  AlignContentProps,
  gridColumn,
  GridColumnProps,
  gridRow,
  GridRowProps,
  gridArea,
  GridAreaProps,
  compose,
} from 'styled-system';
import { ComponentProps } from './utils/types';

const grid = compose(
  systemGrid,
  justifyItems,
  alignItems,
  justifyContent,
  alignContent
);

interface StyledGridProps
  extends SystemGridProps,
    JustifyItemsProps,
    AlignItemsProps,
    JustifyContentProps,
    AlignContentProps {}

export const Grid = styled.div<StyledGridProps>`
  display: grid;
  ${grid};
`;

export type GridProps = ComponentProps<typeof Grid>;

const gridItem = compose(gridColumn, gridRow, gridArea);

interface StyledGridItemProps
  extends GridColumnProps,
    GridRowProps,
    GridAreaProps {}

export const GridItem = styled.div<StyledGridItemProps>`
  ${gridItem}
`;

export type GridItemProps = ComponentProps<typeof GridItem>;
