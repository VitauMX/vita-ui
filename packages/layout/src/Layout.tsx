import * as React from 'react';
import styled from 'styled-components';

const StyledLayout = styled.div`
  background: red;
`;

export interface LayoutProps {}

export const Layout: React.FC<LayoutProps> = () => {
  return <StyledLayout />;
};
