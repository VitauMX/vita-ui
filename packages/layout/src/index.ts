export * from './Grid';
export * from './Box';
export * from './Flex';
