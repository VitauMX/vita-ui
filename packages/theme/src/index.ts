export * from './ThemeProvider';
export * from './theme';
export * from './utils';
