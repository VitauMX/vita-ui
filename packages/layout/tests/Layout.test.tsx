import React from 'react';
import { render } from '@testing-library/react';
import { axe } from 'jest-axe';

import { Grid } from '../src';

describe('Grid', () => {
  it('renders consistently', () => {
    const { container } = render(<Grid />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should have no axe violations', async () => {
    const { container } = render(<Grid />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });

  it('should have display: grid', () => {
    const { container } = render(<Grid />);
    expect(container.firstChild).toHaveStyleRule('display', 'grid');
  });
});
