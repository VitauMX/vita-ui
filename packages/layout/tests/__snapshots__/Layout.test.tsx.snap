// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid renders consistently 1`] = `
.c0 {
  display: grid;
}

<div
  class="c0"
/>
`;
